mission_control/navigator_missions/nav_missions/coral_survey.py

```diff
@@ -20,14 +20,22 @@ def main(navigator):
     result = navigator.fetch_result()
 
     #middle_point = np.array([-10, -70, 0]) 
-    resp = yield navigator.database_query("coral_survey")
-    if not resp.found:
-        result.success = False
-        result.response = result.DbObjectNotFound
-        result.message = "Coral survey not found in the database."
-        defer.returnValue(result)
-
-    middle_point = navigator_tools.point_to_numpy(resp.objects[0].position)
+    est_coral_survey = yield navigator.database_query("Coral_Survey")
+    assert est_coral_survey.found, "Coral Survey ROI not found."
+    
+    yield navigator.move.set_position(est_coral_survey.objects[0]).go()
+
+    totem = yield navigator.database_query("totem")
+    assert totem.found, "No totem objects"
+    
+    # Get the closest totem object to the boat
+    totem_np = map(lambda obj: navigator_tools.point_to_numpy(obj), totem.objects)
+    dist = map(lambda totem_np: np.linalg.norm(totem_np - navigator_tools.point_to_numpy(est_coral_survey.objects[0])), totems_np)
+    
+    # Complain if the buoy is too far away
+    assert min(dist) < 20, "Totems are too far away from Coral Survey"
+    
```

Comment:
Since this is a mission, we shouldn't really error handle this way. 

This is the top of the stack and errors need to be handled here, not thrown, or the mission will not be repeated. (Unless its a MissingPerceptionObject message)
This is the top of the stack and errors need to be handled here, not thrown. (Unless its a MissingPerceptionObject message)